File: frontend/src/types/task.ts
export interface Task {
  id: string
  name: string
  status: 'pending' | 'running' | 'completed' | 'failed' | 'cancelled'
  progress: number
  result?: any
  error?: string
  createdAt: string
  updatedAt?: string
  startedAt?: string
  completedAt?: string
  type: 'analysis' | 'convert' | 'test' | 'git_clone' | 'git_analyze' | 'git_history' | 'git_diff' | 'batch'
  description?: string
  params?: any
  filePath?: string
  agentThoughts?: AgentThought[]
}

export interface TaskCreateData {
  type: string
  name: string
  description?: string
  params?: any
}

export interface AgentThought {
  message: string
  timestamp: number
  type: 'thought' | 'action' | 'result'
}

export interface FileInfo {
  id: string
  filename: string
  size: number
  uploadedAt: string
}

export interface TaskCreateRequest {
  type: string
  name: string
  description?: string
  params?: any
}

export interface TaskCreateResponse {
  task_id: string
  message: string
}

export interface TaskListResponse {
  tasks: Task[]
  total: number
}

export interface TaskResponse {
  task: Task
}

export interface TaskResultResponse {
  task_id: string
  result: any
  status: string
}

export interface CodeAnalysisResult {
  issues: CodeIssue[]
  metrics: CodeMetrics
  suggestions: CodeSuggestion[]
}

export interface CodeIssue {
  line: number
  column: number
  severity: 'error' | 'warning' | 'info'
  message: string
  rule: string
}

export interface CodeMetrics {
  lines: number
  complexity: number
  maintainability: number
  testCoverage?: number
}

export interface CodeSuggestion {
  line: number
  original: string
  suggested: string
  reason: string
}

export interface ConversionResult {
  original: string
  converted: string
  diff: string
  warnings: string[]
}

export interface TestGenerationResult {
  testCode: string
  testFramework: string
  coverage: number
  testCases: TestCase[]
}

export interface TestCase {
  name: string
  description: string
  code: string
}

File: backend-go/internal/models/task.go
package models

import (
	"time"
)

// TaskStatus 任务状态
type TaskStatus string

const (
	TaskStatusPending   TaskStatus = "pending"
	TaskStatusRunning   TaskStatus = "running"
	TaskStatusCompleted TaskStatus = "completed"
	TaskStatusFailed    TaskStatus = "failed"
	TaskStatusCancelled TaskStatus = "cancelled"
)

// TaskType 任务类型
type TaskType string

const (
	TaskTypeAnalysis   TaskType = "analysis"
	TaskTypeConvert    TaskType = "convert"
	TaskTypeTest       TaskType = "test"
	TaskTypeBatch      TaskType = "batch"
	TaskTypeGitClone   TaskType = "git_clone"
	TaskTypeGitAnalyze TaskType = "git_analyze"
	TaskTypeGitHistory TaskType = "git_history"
	TaskTypeGitDiff    TaskType = "git_diff"
)

// Task 任务模型
type Task struct {
	ID          string                 `json:"id"`
	Type        TaskType               `json:"type"`
	Status      TaskStatus             `json:"status"`
	Name        string                 `json:"name"`
	Description string                 `json:"description"`
	Params      map[string]interface{} `json:"params"`
	Result      map[string]interface{} `json:"result,omitempty"`
	Error       string                 `json:"error,omitempty"`
	Progress    int                    `json:"progress"` // 0-100
	CreatedAt   time.Time              `json:"created_at"`
	UpdatedAt   time.Time              `json:"updated_at"`
	StartedAt   *time.Time             `json:"started_at,omitempty"`
	CompletedAt *time.Time             `json:"completed_at,omitempty"`
}

// TaskCreateRequest 任务创建请求
type TaskCreateRequest struct {
	Type        TaskType               `json:"type"`
	Name        string                 `json:"name"`
	Description string                 `json:"description"`
	Params      map[string]interface{} `json:"params"`
}

// TaskCreateResponse 任务创建响应
type TaskCreateResponse struct {
	TaskID  string `json:"task_id"`
	Message string `json:"message"`
}

// TaskResponse 任务响应
type TaskResponse struct {
	Task Task `json:"task"`
}

// TaskListResponse 任务列表响应
type TaskListResponse struct {
	Tasks []Task `json:"tasks"`
	Total int    `json:"total"`
}

// TaskResultResponse 任务结果响应
type TaskResultResponse struct {
	TaskID string                 `json:"task_id"`
	Result map[string]interface{} `json:"result"`
	Status TaskStatus             `json:"status"`
}